### PR TITLE
✏️qawolf.create()

### PR DESCRIPTION
**Pull request by jperl**

### CODE REVIEW

File: src/build-code/buildTemplate.ts [hunk 1]
```diff
@@ -52,6 +52,7 @@ if (require.main === module) {
   (async () => {
     const browser = await qawolf.launch();
     ${buildNewContext(device)}
+    await qawolf.register(context);
```
Comment (flaurida) [inline src/build-code/buildTemplate.ts]: should we add a small comment here on what this does? just a thought

Comment (jperl) [reply src/build-code/buildTemplate.ts]: We could link to our docs?

File: src/create-code/CodeFileUpdater.ts [hunk 2]
```diff
@@ -1,22 +1,54 @@
...
 
-  protected async updateCode(code: string): Promise<void> {
+  protected async _updateCode(code: string): Promise<void> {
     await outputFile(this._path, code, 'utf8');
   }
+
+  public async discard(): Promise<void> {
+    this._locked = true;
+
+    if (process.env.QAW_DISCARD === '1') {
```
Comment (flaurida) [inline src/create-code/CodeFileUpdater.ts]: assuming default here should be to revert

File: src/create-code/CreateManager.ts [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    await Promise.all([
+      this._codeUpdater.update({ steps }),
+      this._selectorUpdater.update({ steps }),
+    ]);
+  }
+
+  public async finalize(): Promise<void> {
+    const shouldSave = await promptSaveRepl(this._codeUpdater.path());
+    if (shouldSave) {
+      await this._codeUpdater.finalize();
```
Comment (flaurida) [inline src/create-code/CreateManager.ts]: what about `_selectorUpdater`

File: test/create-code/__snapshots__/CodeFileUpdater.test.ts.snap [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  // 🐺 CREATE CODE HERE
+  otherCode();"
+`;
+
+exports[`CodeFileUpdater update saves code with new steps 2`] = `
+"  someCode();
+  await page.click(selectors[0]);
+  await page.click(selectors[1]);
+  await page.type(selectors[2], \\"tomsmith\\");
+  await page.type(selectors[3], \\"↓Tab↑Tab\\");
```
Comment (jperl) [inline test/create-code/__snapshots__/CodeFileUpdater.test.ts.snap]: This will change after #441

File: src/create-code/CreateManager.ts (hunk 3 above)
Comment (jperl) [reply src/create-code/CreateManager.ts]: There is nothing to finalize

File: src/create-code/CodeFileUpdater.ts (hunk 2 above)
Comment (jperl) [reply src/create-code/CodeFileUpdater.ts]: This is going to be a hidden env variable we pass from `npx qawolf create` to know to remove the file.

**Pull request merged**